### ISSUE TITLE
setDate() is not accepting a Date object.

### ISSUE THREAD

**Issue by trusktr** [eternicode/bootstrap-datepicker#1316]
For example, the following doesn't work, resulting in `getDate()` returning `null`.

```js
    $el.datepicker('setDate', new Date()); // should intuitively work
```

But the following *does* work.

```js
    $el.datepicker('setDate', (new Date()).toISOString());
```

I'm instantiating like this:

```js
    $el = $('[data-js=datepicker]').datepicker({
      format: 'mm-dd-yyyy',
      autoclose: true,
      endDate: endDate,
      todayBtn: "linked",
      todayHighlight: true
    });
```

and the markup looks something like this:

```html
  <div class="input-group date col-md-6" data-js="datepicker">
    <label for="date-input">Agent</label>
    <input id="date-input"
      name="date" class="form-control datepicker"
      type="text" placeholder="mm-dd-yyyy"/>
    <span class="input-group-addon"><i class="fui-calendar"></i></span>
  </div>
```

I've experienced this in v1.3.1 and v1.5.0. I haven't tried other versions yet.

**Comment by trusktr** [comment]
In the following fiddle, it *does* work, but I'm not sure what version of datepicker it's using:

http://jsfiddle.net/DwT8M/413/

**Comment by trusktr** [comment]
Here's a reproduction in jsfiddle, showing that today's date isn't selected (though the date *does* appear in the input unlike in my actual case):

http://jsfiddle.net/DwT8M/415/

**Comment by trusktr** [comment]
Here's a reproduction in jsfiddle, showing that today's date isn't selected (though the date does appear in the input unlike in my actual case):

http://jsfiddle.net/DwT8M/416/

**Comment by trusktr** [comment]
Okay, so calling `update()` does update the selection, so that's fine and my (client's proprietary) code is working now. Upgrading from 1.3.1 to 1.5.0 seems to have fixed a problem where the `changeDate` wasn't always firing after calling `setDate`. Anyway, it's all working now.

**Comment by acrobat** [comment]
Good that the problem is solved @trusktr, but keep in mind that 1.5.0 is still in development! 1.4.0 is the latest stable release
